src/main/java/org/wso2/carbon/apimgt/securityenforcer/publisher/HttpDataPublisher.java

```diff
@@ -0,0 +1,199 @@
+        setEndPoint(aseConfiguration.getEndPoint());
+    }
+
+    public HttpDataPublisher(String endPoint, String authToken) {
+        setAuthToken(authToken);
+        setEndPoint(endPoint);
+    }
+
+    public AseResponseDTO publish(JSONObject data, String correlationID, String resource) {
+        HttpPost postRequest = new HttpPost(endPoint + "/ase/" + resource);
+        postRequest.addHeader(AISecurityHandlerConstants.ASE_TOKEN_HEADER, authToken);
+        postRequest.addHeader(AISecurityHandlerConstants.X_CORRELATION_ID_HEADER, correlationID);
+
+        CloseableHttpResponse response = null;
+        AseResponseDTO aseResponseDTO = null;
+
+        try {
+            postRequest.setEntity(new StringEntity(data.toString()));
+            long publishingStartTime = System.nanoTime();
+            response = httpClient.execute(postRequest);
+            long publishingEndTime = System.nanoTime();
+
+            if (response != null) {
+                aseResponseDTO = new AseResponseDTO();
+                aseResponseDTO.setResponseMessage(response.getStatusLine().getReasonPhrase());
+                aseResponseDTO.setResponseCode(response.getStatusLine().getStatusCode());
+
+                switch (response.getStatusLine().getStatusCode()) {
+                case AISecurityHandlerConstants.ASE_RESPONSE_CODE_INCORRECT_JSON:
+                    log.error("Incorrect JSON format sent for the ASE from the request " + correlationID);
+                    break;
+                case AISecurityHandlerConstants.ASE_RESPONSE_CODE_UNKNOWN_API:
+                    log.error("Unknown API for the request " + correlationID);
+                    break;
+                case AISecurityHandlerConstants.ASE_RESPONSE_CODE_UNAUTHORIZED:
+                    log.error("Authentication failure (ASE-Token) for the request " + correlationID);
+                    break;
+                }
+
+                if (log.isDebugEnabled()) {
+                    log.debug("PING ASE Response for method: " + resource + ", correlation ID " + correlationID
+                            + " , response: " + response.toString()
+                            + ", connection time for the request in nano seconds is " + (publishingEndTime
+                            - publishingStartTime));
+                }
+            } else {
+                log.error("Null response returned from ASE for the request " + correlationID);
+            }
+        } catch (SocketTimeoutException e) {
+            log.error(
+                    "Socket timeout exception when sending the HTTP Request with id " + correlationID + " ,Error : " + e
+                            .getMessage());
+            aseResponseDTO = getDefaultAcceptResponse();
+        } catch (SSLHandshakeException e) {
+            log.error("SSLHandshakeException exception when sending the HTTP Request with id " + correlationID
+                    + " ,Error : " + e.getMessage());
```

Comment:
The entire SSLHandshakeException needs to be logged here. Only exception message is being currently logged